### ISSUE TITLE
On iOS simulator and physical device, sometimes I get this when closing popup:

### ISSUE THREAD

**Issue by scastria** [rotorgames/Rg.Plugins.Popup#93]
I doesn't happen all the time, but enough.  About once every 10 times.  I have not found a pattern to make it happen all the time.  I should also add that my popup contains an Entry and I did not want the user to have to click in the entry to get the keyboard, therefore, in the OnAppearing of my popup page, I am setting the focus on my entry.  This causes the popup page and keyboard to appear at the same time.  Maybe that is related??

```
2017-03-13 09:50:12.053 FDCapture.iOS[830:42482] Can't find keyplane that supports type 8 for keyboard iPhone-PortraitChoco-DecimalPad; using 2876297567_PortraitChoco_iPhone-Simple-Pad_Default
Thread finished: <Thread Pool> #5
2017-03-13 09:50:24.923 FDCapture.iOS[830:42482] Can't find keyplane that supports type 8 for keyboard iPhone-PortraitChoco-DecimalPad; using 2876297567_PortraitChoco_iPhone-Simple-Pad_Default
Thread finished: <Thread Pool> #11
Thread finished: <Thread Pool> #8
Thread started: <Thread Pool> #13
2017-03-13 09:50:29.023 FDCapture.iOS[830:42482] Can't find keyplane that supports type 8 for keyboard iPhone-PortraitChoco-DecimalPad; using 2876297567_PortraitChoco_iPhone-Simple-Pad_Default
2017-03-13 09:50:42.777 FDCapture.iOS[830:42482] Can't find keyplane that supports type 8 for keyboard iPhone-PortraitChoco-DecimalPad; using 2876297567_PortraitChoco_iPhone-Simple-Pad_Default
2017-03-13 09:50:45.376 FDCapture.iOS[830:43619] critical: Stacktrace:

2017-03-13 09:50:45.376 FDCapture.iOS[830:43619] critical:   at <unknown> <0xffffffff>
2017-03-13 09:50:45.377 FDCapture.iOS[830:43619] critical:   at (wrapper managed-to-native) ObjCRuntime.Messaging.void_objc_msgSend_float (intptr,intptr,single) <0x00012>
2017-03-13 09:50:45.382 FDCapture.iOS[830:43619] critical:   at CoreAnimation.CALayer.set_Opacity (single) [0x0000b] in /Users/builder/data/lanes/3969/7beaef43/source/xamarin-macios/src/build/ios/native/CoreAnimation/CALayer.g.cs:1770
2017-03-13 09:50:45.383 FDCapture.iOS[830:43619] critical:   at Xamarin.Forms.Platform.iOS.VisualElementTracker/<>c__DisplayClass23_0.<OnUpdateNativeControl>b__0 () [0x0015a] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.iOS\VisualElementTracker.cs:178
2017-03-13 09:50:45.383 FDCapture.iOS[830:43619] critical:   at Xamarin.Forms.Platform.iOS.CADisplayLinkTicker.StartThread () [0x0001d] in C:\BuildAgent3\work\ca3766cfc22354a1\Xamarin.Forms.Platform.iOS\CADisplayLinkTicker.cs:57
2017-03-13 09:50:45.384 FDCapture.iOS[830:43619] critical:   at System.Threading.ThreadHelper.ThreadStart_Context (object) [0x00017] in /Users/builder/data/lanes/3969/7beaef43/source/xamarin-macios/_ios-build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/src/mono/mcs/class/referencesource/mscorlib/system/threading/thread.cs:68
2017-03-13 09:50:45.384 FDCapture.iOS[830:43619] critical:   at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext,System.Threading.ContextCallback,object,bool) [0x0008d] in /Users/builder/data/lanes/3969/7beaef43/source/xamarin-macios/_ios-build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/src/mono/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:957
2017-03-13 09:50:45.384 FDCapture.iOS[830:43619] critical:   at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext,System.Threading.ContextCallback,object,bool) [0x00000] in /Users/builder/data/lanes/3969/7beaef43/source/xamarin-macios/_ios-build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/src/mono/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:904
2017-03-13 09:50:45.385 FDCapture.iOS[830:43619] critical:   at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext,System.Threading.ContextCallback,object) [0x00031] in /Users/builder/data/lanes/3969/7beaef43/source/xamarin-macios/_ios-build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/src/mono/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:893
2017-03-13 09:50:45.385 FDCapture.iOS[830:43619] critical:   at System.Threading.ThreadHelper.ThreadStart () [0x0000b] in /Users/builder/data/lanes/3969/7beaef43/source/xamarin-macios/_ios-build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/src/mono/mcs/class/referencesource/mscorlib/system/threading/thread.cs:105
2017-03-13 09:50:45.385 FDCapture.iOS[830:43619] critical:   at (wrapper runtime-invoke) object.runtime_invoke_void__this__ (object,intptr,intptr,intptr) <IL 0x0004f, 0x001d0>
2017-03-13 09:50:45.385 FDCapture.iOS[830:43619] critical: 
Native stacktrace:

2017-03-13 09:50:45.385 FDCapture.iOS[830:43619] critical: 
=================================================================
Got a SIGSEGV while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries 
used by your application.
=================================================================
```

**Comment by scastria** [comment]
You can see that it is crashing in CoreAnimation.  Therefore, I thought I would try turning off the PopupPage animations for push and pop and that seemed to work.  So I think there is either a bug in this Rg.Plugins.Popup component or a bug in the underlying Xamarin wrapper around CoreAnimation or in Xamarin.Forms VisualElementTracker stuff (which is also in the stack trace).

**Comment by scastria** [comment]
Xamarin's response from looking at the stack trace was wondering if anything was being done OFF the main UI thread?

**Comment by rotorgames** [comment]
@scastria This happens only with Popup plugin or you saw it with other Pages?

**Comment by scastria** [comment]
I have never seen this with any other pages.  My first encounter with this stack trace was with this popup plugin.  and it only seems to happen on the dismiss, not the popup.

**Comment by rotorgames** [comment]
@scastria I think that this problem is not related with my plugin. I don't see code from my plugin in stack trace. I think that you should be looking for problem in your code or in xamarin forms code.

**Comment by scastria** [comment]
Just because your code is not in the stack trace does not mean the problem is not in your code.  For example, I can put a bad class cast in my code and get my XF app to fail with a stack trace that does not include my code where the bad class cast is.

At all boils down to:

await Navigation.PopAllPopupAsync(true);

causes the crash, but:

await Navigation.PopAllPopupAsync(false);

does NOT.  I will see if I can create a sample program.

**Comment by rotorgames** [comment]
@scastria Ok. Create example solution and I will look it

**Comment by rotorgames** [comment]
Also see #94

**Comment by renfred** [comment]
Came across this issue the other day, any solution yet?

**Comment by rotorgames** [comment]
This issue was fixed in v1.1.0-pre1. Please test it and tell me about your tests. Thank you.
